protocol/common/raw_messages_persistence.go

```diff
@@ -205,50 +204,39 @@ func (db RawMessagesPersistence) MarkAsConfirmed(dataSyncID []byte, atLeastOne b
 	}
 
 	// Select any tuple that has a message_id with a datasync_id = ? and that has just been confirmed
-	rows, err := tx.Query(`SELECT message_id,confirmed_at FROM raw_message_confirmations WHERE message_id IN (SELECT message_id FROM raw_message_confirmations WHERE datasync_id = ?)`, dataSyncID)
+	rows, err := tx.Query(`SELECT message_id,confirmed_at FROM raw_message_confirmations WHERE message_id = (SELECT message_id FROM raw_message_confirmations WHERE datasync_id = ? LIMIT 1)`, dataSyncID)
```

Comment:
Can we simplify this to:
`SELECT message_id,confirmed_at FROM raw_message_confirmations WHERE datasync_id = ?`
?

Reply:
I don't think that would work, essentially if you take a group chat message we will have 
3 identical `messageID` and 3 different `datasyncID`.
We need to check that all 3 have been confirmed (depending on `atLeastOne`, but suppose `atLeatOne` is `false` for now), basically make sure that all of them have been confirmed.

Reply:
Ah, I got the idea, thanks for explanation!